agent/state.go

```diff
@@ -9,18 +9,29 @@ import (
 	"sync"
 )
 
-//go:generate stringer -type=state
-type state uint8
+//go:generate stringer -type=State
+type State uint8
 
 const (
-	idle state = iota
-	receivingManifest
-	receivingAlgorithm
-	receivingData
-	running
-	resultsReady
-	complete
-	failed
+	Idle State = iota
+	ReceivingManifest
+	ReceivingAlgorithm
+	ReceivingData
+	Running
+	ResultFetch
+	Complete
+	Failed
+)
+
+//go:generate stringer -type=Status
+type Status uint8
+
+const (
+	IdleState Status = iota
+	InProgress
+	Ready
+	Completed
+	Terminated
```

Comment:
manager also has status and will have a general pattern, started, in progress, complete and failed

Reply:
I am using these statuses for manager.